### DISCUSSION TITLE
Welcome to discovery-mb2 Discussions!

### DISCUSSION THREAD

**Discussion by BartMassey** [rust-embedded/discovery-mb2#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Like all our activities, discussions here are subject to the [Rust Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct).** We take this quite seriously: please do likewise.

**Comment by Kerssss** [comment]
Hello,

Today I started reading the book for microbit controller. I was able to do everything but I got stuck at section 5.4. When executing the command 'target remote :1337' I do not get the same reply as described in the book:

- In the book:
`
(gdb) target remote :1337
Remote debugging using :1337
0x00000116 in nrf52833_pac::{{impl}}::fmt (self=0xd472e165, f=0x3c195ff7) at /home/nix/.cargo/registry/src/github.com-1ecc6299db9ec823/nrf52833-pac-0.9.0/src/lib.rs:157
157     #[derive(Copy, Clone, Debug)]
`

- In my PC:
`
(gdb) target remote :1337
Remote debugging using :1337
warning: while parsing target description (at line 4): Target description specified unknown architecture "armv7"
warning: Could not load XML target description; ignoring
0x00000000 in __vector_table ()
`

I do not know if this might have something to do but when using breakpoints, I  am not able to print the value of x:

`
(gdb) break 15
Breakpoint 4 at 0x172: file mdbook\src\05-meet-your-software\examples/init.rs, line 15.
(gdb) print x
No symbol "x" in current context.
(gdb) print(x)
No symbol "x" in current context.
(gdb) print _y
No symbol "_y" in current context.
(gdb) print &x
No symbol "x" in current context.
`

And the loaded program is:

`
#![deny(unsafe_code)]
#![no_main]
#![no_std]

use cortex_m::asm;
use cortex_m_rt::entry;
use microbit as _;
use panic_halt as _;

#[entry]
fn main() -> ! {
    #[allow(clippy::needless_late_init)]
    let _y;
    let x = 42;
    _y = x;

    // infinite loop; just so we don't leave this stack frame
    loop {
        asm::nop();
    }
}
`
Thank you!

**Comment by BartMassey** [comment]
Looks like your `gdb` knows nothing of ARM — you probably need `gdb-multiarch` or similar for your platform. Take a look at issues #27 and #65 which have good descriptions, and file something there if you still have trouble.

We really need to get a PR from somebody updating the Setup and this chapter to make sure everybody knows how they need to be set up on each platform. I don't do Windows or Mac, so I am of little help there I'm afraid.
